server/src/test/java/io/crate/expression/udf/UserDefinedFunctionServiceTest.java

```diff
@@ -158,4 +159,22 @@ public void test_validate_partitioned_table_while_dropping_udf() throws Exceptio
             "Cannot drop function 'foo', it is still in use by 'doc.p1.gen AS doc.foo(id)'"
         );
     }
+
+    @Test
+    public void test_validate_sub_columns_while_dropping_udf() throws Exception {
+        UserDefinedFunctionsMetadata metadataWithoutFunction = UserDefinedFunctionsMetadata.of();
+        SQLExecutor executor = SQLExecutor.builder(clusterService)
+            .addUDFLanguage(DUMMY_LANG)
+            .addUDF(FOO)
+            .addPartitionedTable("create table doc.p1 (o object as (id int), gen as foo(o['id'])) partitioned by (o['id'])")
+            .build();
+
+        assertThatThrownBy(() -> executor.udfService().validateFunctionIsNotInUseByGeneratedColumn(
+                               Schemas.DOC_SCHEMA_NAME,
+                               "foo",
+                               metadataWithoutFunction,
+                               clusterService.state()))
+            .hasMessageContaining("Cannot drop function 'foo', it is still in use by 'doc.p1.gen AS doc.foo(o['id'])'")
+            .isInstanceOf(IllegalArgumentException.class);
```

Comment:
```suggestion
            .isExactlyInstanceOf(IllegalArgumentException.class);
```
and could you please move it one line above before asserting the message (just for consistency)

Reply:
Done, Thank you